src/internal.c

```diff
@@ -18480,11 +18492,12 @@ static int DoServerKeyExchange(WOLFSSL* ssl, const byte* input,
                                 args->sigSz = (word16)ret;
                                 ret = 0;
                             }
-
-                            /* peerRsaKey */
-                            FreeKey(ssl, DYNAMIC_TYPE_RSA,
+                            if (ret == 0) {
```

Comment:
should this change be exclusive to ASYNC? It changes when the peer RSA key gets free'd in all cases.

Reply:
I had it this way then saw some other code Sean had done to free keys early that uses this logic. It could be `ret != WC_PENDING_E` here although It does get free'd in the cleanup for failure cases below. I'll adjust these cases.